turbine/src/quic_endpoint.rs

```diff
@@ -197,60 +246,83 @@ async fn handle_connecting(
     endpoint: Endpoint,
     connecting: Connecting,
     sender: Sender<(Pubkey, SocketAddr, Bytes)>,
-    cache: Arc<RwLock<ConnectionCache>>,
+    router: Arc<AsyncRwLock<HashMap<SocketAddr, AsyncSender<Bytes>>>>,
+    cache: Arc<Mutex<HashMap<Pubkey, Connection>>>,
 ) -> Result<(), Error> {
     let connection = connecting.await?;
     let remote_address = connection.remote_address();
     let remote_pubkey = get_remote_pubkey(&connection)?;
-    handle_connection_error(
+    let receiver = {
+        let (sender, receiver) = tokio::sync::mpsc::channel(ROUTER_CHANNEL_BUFFER);
+        router.write().await.insert(remote_address, sender);
+        receiver
+    };
```

Comment:
Why is the server task always creating a new channel, while the client task reuses one if it exists? Is it possible for the client side to have already tried to initiate a connection to that remote address? If that's the case, it looks like the server side would be clobbering the previous channel

Reply:
The server side does not initiate a connection, it only accepts incoming connections from remote nodes.

If there is already a connection and for whatever reason the remote node initiates a new connection, then yes, it will drop the previous connection and replace it with the new one. This happens both in the `router` hash-map here, and the `cache`:
https://github.com/solana-labs/solana/blob/dc3c82729/turbine/src/quic_endpoint.rs#L407-L409

We can possibly allow multiple connections per pubkey by having a `Vec<Connection>` instead of a single `Connection`, but for now I think a single `Connection` per pubkey would be simpler.

Reply:
Ok great, that explains it, thanks! No need to have a `Vec<Connection>` -- resetting on a new remote connection makes sense.